config/backend-config/multitenant_workspaces.go

```diff
@@ -143,6 +143,9 @@ func (wc *multiTenantWorkspacesConfig) getFromAPI(ctx context.Context, _ string)
 		}
 		sourcesJSON.Sources = append(sourcesJSON.Sources, workspaceConfig.Sources...)
 	}
+	sourcesJSON.ConnectionFlags.URL = config.GetEnv("CP_ROUTER_URL", "")
```

Comment:
Instead of using `config.GetEnv` directly inside a method that shouldn't depend nor be aware of `config` or environment variables, we should use struct attributes that are way easier to tests and that decouple the struct from the above. Also it is best not to spread "construction" logic everywhere. I think we should move `config.GetEnv("CP_ROUTER_URL, "")` in the `loadConfig()` function and then have the constructor (i.e. `newForDeployment`) inject the `cpRouterURL` into the `multiTenantWorkspacesConfig`:

```go
backendConfig.workspaceConfig = &multiTenantWorkspacesConfig{
	configBackendURL: parsedConfigBackendURL,
	configEnvHandler: configEnvHandler,
	cpRouterURL: cpRouterURL,
}
```

Reply:
Done